### PR TITLE
Update formula_templater to swap 'plist' for 'service'

### LINKED ISSUE #201: Calling plist_options is deprecated
got the following message when calling `brew doctor`
```
Warning: Calling plist_options is deprecated! Use service.require_root instead.
Please report this issue to the hashicorp/tap tap (not Homebrew/brew or Homebrew/core), or even better, submit a PR to fix it:
  /opt/homebrew/Library/Taps/hashicorp/homebrew-tap/Formula/vault.rb:37
```

### PR DESCRIPTION
**Pull request by jlucktay**
The use of `plist_options` is deprecated by Homebrew, and causes `brew` to throw warnings such as the following:

image

c9f4f14da11626fe4e9966cab22dddaa11bc072a & 9a2cfead87a0f7464a00aa7033edd4d882e33ecd: Updated the expected test output and the generated output from the template in `formula_templater` to the `service` block instead of `plist`.

60d1ebda132da66c5a6d900d8ec9ead612eea14d: Ran [`gofumpt`](https://github.com/mvdan/gofumpt) and fixed some basic lint issues in the `formula_templater` code.

6cce3f3927c3a87a9c3bbee03719b0a59746094a: Updated Go version and bumped dependencies for `formula_templater`.

18d7956e1d3772bfb2749cfa979063a55f28e8ea: Aligned checksums with `SHA256SUMS` files on https://releases.hashicorp.com.

e3241a2f228a43c143eb748c265233eddb20a69c: Ran `formula_templater <service> <version> config.hcl` on all formulas using `plist`.

Fixes #201.

### CODE REVIEW

File: Formula/consul-enterprise.rb [hunk 1]
```diff
@@ -34,40 +34,12 @@ def install
...
-    <string>#{var}/log/consul.log</string>
-    <key>StandardOutPath</key>
-    <string>#{var}/log/consul.log</string>
-</dict>
-</plist>
-
-EOS
+  service do
+    run [bin/"consul", "agent", "-dev", "-bind", "127.0.0.1"]
+    keep_alive true
```
Comment (yasintahaerol) [inline Formula/consul-enterprise.rb:39]: Hey, is this has to be like that `keep_alive successful_exit: false` and also for others to.

Comment (yasintahaerol) [reply Formula/consul-enterprise.rb:39]: No one line. I mean like this [Link](https://github.com/buildkite/homebrew-buildkite/pull/31/files)

Comment (jlucktay) [reply Formula/consul-enterprise.rb]: a5a0c454b4ba43b02ad07fc0058e964f84b502a6 should resolve this.

**Comment by jlucktay** [comment]
Ping some HashiCorp folks with recent activity here: @CalebAlbers @mdeggies @claire-labry - any chance of a review/merge please? 🙏

**Comment by bennetfabian** [comment]
@jlucktay Is this ready to be merged?
@radeksimko Could you take a look at this?

The warning message is not the end of the world but of course it's a bit annoying

**Comment by jlucktay** [comment]
> @jlucktay Is this ready to be merged?

Hi @bennetfabian - yes, this is ready to merge. 🚀

**Comment by mdeggies** [comment]
Hey @jlucktay, thanks for your contribution! The team is reviewing this and will get back to you with feedback shortly.

**Pull request merged**